catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsMetacardSupport.java

```diff
@@ -94,15 +94,20 @@ void generateMetacardAndContentItems(List<ContentItem> incomingContentItems,
         for (ContentItem contentItem : incomingContentItems) {
             try {
                 Path tmpPath = null;
+                String fileName;
                 long size;
                 try (InputStream inputStream = contentItem.getInputStream()) {
+                    fileName = contentItem.getFilename();
                     if (inputStream == null) {
                         throw new IngestException(
                                 "Could not copy bytes of content message.  Message was NULL.");
                     }
 
-                    String sanitizedFilename =
-                            InputValidation.sanitizeFilename(contentItem.getFilename());
+                    if (!InputValidation.checkForClientSideVulnerableFileName(fileName)) {
+                        throw new IngestException("Ignored filename found.");
```

Comment:
Maybe I'm losing some context in the scope of the diff, but is this failing the ingest here? The documentation says that these names will be sanitized if they're in the `bad.files` or `bad.file.extensions` lists. Only the `bad.mime.types` should be rejected